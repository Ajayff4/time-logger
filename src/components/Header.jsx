import React from 'react'

export default function Header() {
    return (
        <header id="headerComponent">
            <div className="header" id="header">
                <center>
                    <h1 id="titleText">Time Logger App</h1>
                </center>
            </div>  
        </header>      
    )
}

// data-test: attribute is used to specify its test attribute.