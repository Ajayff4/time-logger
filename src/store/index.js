import { createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(rootSaga)

export default store