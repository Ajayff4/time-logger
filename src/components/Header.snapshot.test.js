import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Header Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><Header /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});