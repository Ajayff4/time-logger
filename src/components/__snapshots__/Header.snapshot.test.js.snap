// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Snapshot Testing... renders component correctly 1`] = `
<header
  id="headerComponent"
>
  <div
    className="header"
    id="header"
  >
    <center>
      <h1
        id="titleText"
      >
        Time Logger App
      </h1>
    </center>
  </div>
</header>
`;
